addons/account_voucher/migrations/9.0.1.0/post-migration.py

```diff
@@ -4,16 +4,102 @@
 from openupgradelib import openupgrade
 
 
-@openupgrade.migrate()
-def migrate(cr, version):
+def get_ref(cr, module, xml_id):
+    """Get reference (res_id) from module and xml_id from ir_model_data."""
+    cr.execute("""
+        SELECT res_id
+        FROM ir_model_data
+        WHERE module = %s AND name = %s
+    """, (module, xml_id))
+    return cr.fetchone()[0]
+
+
+def migrate_voucher_types(cr):
+    """Migrate voucher types."""
+    # TODO: Should we really keep receipt and payment records? Might contain
+    # duplicate info with the sales and purchases for which they have been
+    # made, especially now that we migrate to account.payment.
     openupgrade.map_values(
         cr, openupgrade.get_legacy_name('type'), 'voucher_type',
         [('receipt', 'sale'), ('payment', 'purchase')],
         table='account_voucher')
+
+def create_voucher_line_tax_lines(cr):
+    """Migrate tax information on voucher lines to m2m relation."""
     cr.execute(
         "insert into account_tax_account_voucher_line_rel "
         "(account_tax_id, account_voucher_line_id) "
         "select v.tax_id, l.id "
         "from account_voucher v "
         "join account_voucher_line l on l.voucher_id=v.id "
         "where v.tax_id is not null")
+
+def create_payments_from_vouchers(cr):
+    """Create account.payment rows from appropiate account.voucher rows."""
+    receipt_method = get_ref(
+        cr, 'account', 'account_payment_method_manual_in'
+    )
+    payment_method = get_ref(
+        cr, 'account', 'account_payment_method_manual_out'
+    )
+    # Keep original id, to later migrate references from other tables,
+    # although it seems not te be needed for now.
+    # Possible states of account.payment: draft, posted, sent, reconciled
+    # Possible states of account_voucher: draft, cancel, proforma, posted
+    # "type" field in 8.0 renamed to "voucher_type" during migration.
+    # Field destination_journal_id is used for internal transfer payments.
+    # Not used here.
+    # As far as I can see there is no connection between invoices or moves
+    # in 8.0 taht we can or have to migrate to 9.0.
+    cr.execute("""
```

Comment:
I think using plain SQL here is much faster, but above all more clear/clean/readable.